pkg/ocm/helpers.go

```diff
@@ -655,3 +656,77 @@ func ValidateAvailabilityZonesCount(multiAZ bool, availabilityZonesCount int) er
 
 	return nil
 }
+
+func (c *Client) GetVersion(userRequestedVersion string, channelGroup string) (string, error) {
+	versionList, err := c.GetVersionsList(channelGroup)
+	if err != nil {
+		err := fmt.Errorf("%v", err)
+		return userRequestedVersion, err
+	}
+
+	if userRequestedVersion == "" {
+		return versionList[0], nil
+	}
+
+	hasVersion := false
+	for _, vs := range versionList {
+		if vs == userRequestedVersion {
+			hasVersion = true
+			break
+		}
+	}
+
+	if !hasVersion {
+		versionSet := make(map[string]bool)
+		for _, vs := range versionList {
+			versionSet[vs] = true
+		}
+		err := fmt.Errorf("a valid version number must be specified\nValid versions: %v", KeysString(versionSet))
```

Comment:
i dont know how many valid versions we will get to list . can i see the sample

Reply:
After filtering everything the final list will show up as something like this:
`Valid versions: [4.10, 4.9, 4.8, 4.7, 4.11]`

As we are mapping versions to 'major.minor' and setting up a set of them, but the original response depends on the channel group. Nightly has a format like` "openshift-v4.11.0-0.nightly-2022-09-15-062234-nightly"` and around 40 entries. But stable has more than 100